### PR TITLE
Extract Conn interface and add CompositeConn implementation for interacting with services exposed directly rather than via MMD

### PR DESCRIPTION
**Pull request by ryant1986**

### CODE REVIEW

File: mmd/compositeconn.go [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+		}
+	}
+
+	if service == serviceDiscoveryServiceName {
+		return ISTIO, nil
+	}
+
+	serviceDiscoveryServiceConn, err := c.getOrCreateConnection(serviceDiscoveryServiceName)
+	if err != nil {
+		return -1, err
```
Comment (pkuprys) [inline mmd/compositeconn.go:287]:
```suggestion
		return ERROR, err
```

File: mmd/compositeconn.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	}
+
+	resp, err := serviceDiscoveryServiceConn.Call(serviceDiscoveryServiceName, map[string]interface{}{"service": service})
+	if err != nil {
+		return -1, err
```
Comment (pkuprys) [inline mmd/compositeconn.go:292]:
```suggestion
		return ERROR, err
```

File: mmd/compositeconn.go [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+	}
+
+	respMap, ok := resp.(map[interface{}]interface{})
+	if !ok {
+		return -1, fmt.Errorf("service discovery service response was not a map: %s", reflect.TypeOf(resp))
```
Comment (pkuprys) [inline mmd/compositeconn.go:297]:
```suggestion
		return ERROR, fmt.Errorf("service discovery service response was not a map: %s", reflect.TypeOf(resp))
```

**Review by pkuprys — CHANGES_REQUESTED** [review]
missed a few ERROR accessMethod returns

File: mmd/compositeconn.go (hunk 1 above)
Comment (ryant1986) [reply mmd/compositeconn.go]: done

File: mmd/compositeconn.go (hunk 2 above)
Comment (ryant1986) [reply mmd/compositeconn.go]: done

File: mmd/compositeconn.go (hunk 3 above)
Comment (ryant1986) [reply mmd/compositeconn.go]: done

**Review by pkuprys — APPROVED** [review]
looks good

**Review by jlobraco — APPROVED** [review]
lgtm

**Pull request merged**